### PR TITLE
[BEAM-7102] Adding `jar_packages` experiment option for Python SDK

### PR DESCRIPTION
**Pull request by ihji**
Adding `jar_packages` experiment option for Python SDK

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by ihji** [comment]
R: @lukecwik, @chamikaramj

File: sdks/python/apache_beam/options/pipeline_options.py [hunk 1]
```diff
@@ -777,6 +777,18 @@ def _add_argparse_args(cls, parser):
          'job submission, the files will be staged in the staging area '
          '(--staging_location option) and the workers will install them in '
          'same order they were specified on the command line.'))
+    parser.add_argument(
```
Comment (lukecwik) [inline sdks/python/apache_beam/options/pipeline_options.py]: Shouldn't the environment that contains the Java SDK harness already contain itself and all of the necessary dependencies to execute the cross language IO transform?

Comment (lukecwik) [reply sdks/python/apache_beam/options/pipeline_options.py]: Currently under discussion within https://lists.apache.org/thread.html/6fcee7047f53cf1c0636fb65367ef70842016d57effe2e5795c4137d@%3Cdev.beam.apache.org%3E

**Comment by robertwb** [comment]
If we go with this, can we at least rename it something clearly transient/experimental (or make it an experiment)?

**Comment by chamikaramj** [comment]
+1. How about adding a "CrossLanguagePipelineOptions" object and an option "jar_package_experimental" ?

**Comment by ihji** [comment]
`jar_packages` is now an experimental option.

File: sdks/python/apache_beam/runners/portability/stager.py [hunk 2]
```diff
@@ -308,6 +318,56 @@ def _download_file(from_url, to_path):
   def _is_remote_path(path):
     return path.find('://') != -1
 
+  def _stage_jar_packages(self, jar_packages, staging_location, temp_dir):
+    """Stages a list of local jar packages for Java SDK Harness.
+
+    :param jar_packages: Ordered list of local paths to jar packages to be
+      staged. Only packages on localfile system and GCS are supported.
+    :param staging_location: Staging location for the packages.
```
Comment (chamikaramj) [inline sdks/python/apache_beam/runners/portability/stager.py]: Please add a test

Comment (ihji) [reply sdks/python/apache_beam/runners/portability/stager.py]: Added. Thanks!

**Comment by chamikaramj** [comment]
LGTM. Thanks.

**Pull request merged**